encounter.md

```diff
@@ -0,0 +1,258 @@
+     `404 Not Found` status if encounter provider not exists. If you are not logged in to perform this action, a `401 Unauthorized` status
+     returned.
+     
+    ```console
+        GET /encounter/:target_encounter_uuid/encounterprovider/:target_encounter_provider_uuid
+    ```
+### Create an encounter provider sub resource with properties
+
+* To Create an attribute sub resource for a specific visit resource you need to specify below attributes in the request body.
+If user not logged in to perform this action, a `401 Unauthorized` status returned.
+
+    #### Attributes
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *provider* | `Provider_Type UUID` | UUID of a provider currently registered in OpenMRS (required)
+    *encounterRole* | `Encounter_Role UUID` | UUID of encounter role. This is the role provider will participate during this encounter (required)
+    
+    ```console
+        POST encounter/:target_encounter_uuid/encounterprovider 
+        {
+          "provider": "bb1a7781-7896-40be-aaca-7d1b41d843a6",
+          "encounterRole": "240b26f9-dd88-4172-823d-4a8bfeb7841f"
+        }
+    ```
+ 
+ 
+### Update encounter provider sub resource
+
+* Updates an encounter provider sub resource value with given uuid, this method will only modify value of the sub resource.Returns 
+a `404 Not Found` status if encounter provider not exists.If user not logged in to perform this action, a `401 Unauthorized` status
+returned.
+
+    #### Query Parameters
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *parent_uuid* | `target_encounter_uuid` | Target encounter resource UUID
+    *uuid* | `target_encounter_provider_uuid` | Target encounter provider resource UUID
+
+    #### Attributes
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *provider* | `Provider_Type UUID` | UUID of a provider currently registered in OpenMRS (required)
+    *encounterRole* | `Encounter_Role UUID` | UUID of encounter role. This is the role provider will participate during this encounter (required)
+
+    ```console
+        POST encounter/:target_encounter_uuid/encounterprovider/:target_encounter_provider_uuid
+        {
+          "provider": "bb1a7781-7896-40be-aaca-7d1b41d843a6",
+          "encounterRole": "240b26f9-dd88-4172-823d-4a8bfeb7841f"
+        }
+    ```
+### Delete encounter provider sub resource
+
+* Delete or Retire a target encounter provider sub resource by its UUID.Returns a `404 Not Found` status if attribute not exists. 
+If user not logged in to perform this action, a `401 Unauthorized` status returned.
+
+    #### Query Parameters
+
+    Parameter | Type | Description
+    --- | --- | ---
+    *purge* | `Boolean` | The resource will be voided/retired unless purge = ‘true’. Purging will attempt to irreversibly remove the encounter provider type from the system. Encounter provider types that have been used (i.e., are referenced from existing data) cannot be purged.
```

Comment:
Encounters providers are [voided](https://docs.openmrs.org/datamodel/openmrs-refapp-data-model-1.9.html#encounter_provider.voided) (not retired).